src/internal/m365/collection/drive/delta_tree.go

```diff
@@ -65,16 +70,103 @@ type nodeyMcNodeFace struct {
 func newNodeyMcNodeFace(
 	parent *nodeyMcNodeFace,
 	id, name string,
-	prev path.Path,
+	prev path.Elements,
 	isPackage bool,
 ) *nodeyMcNodeFace {
 	return &nodeyMcNodeFace{
 		parent:    parent,
 		id:        id,
 		name:      name,
 		prev:      prev,
-		childDirs: map[string]*nodeyMcNodeFace{},
+		children:  map[string]*nodeyMcNodeFace{},
 		items:     map[string]time.Time{},
 		isPackage: isPackage,
 	}
 }
+
+// ---------------------------------------------------------------------------
+// folder handling
+// ---------------------------------------------------------------------------
+
+func (face *folderyMcFolderFace) SetFolder(
```

Comment:
Yeah, I thought of that.  May revisit as implementation progresses, but wanted to avoid for now for simplicity.  To begin with I wanted to use the method signatures to showcase the exact details about the behavior being handled.